lib/configuration/configuration_parser.js

```diff
@@ -0,0 +1,50 @@
+const Logger = require('../logger');
+const fsPromises = require('fs/promises');
+const clientConfiguration = require('./client_configuration');
+
+exports.parse = async function (configFilePath) {
+  const configFile = await tryToFind(configFilePath);
+  return configFile == null ? null : tryToParse(configFile);
+};
+
+function tryToFind (filePath) {
+  if (!filePath) {
+    return Promise.resolve(null);
+  }
+  return fsPromises.readFile(filePath, { encoding: 'utf8' })
+    .catch(err => {
+      const errorMessage = 'Finding easy logging configuration failed';
+      Logger.getInstance().error(errorMessage + '\n' + err.toString());
+      throw new Error(errorMessage);
+    });
+}
+
+function tryToParse (configurationJson) {
+  try {
+    const parsedConfiguration = JSON.parse(configurationJson);
+    validate(parsedConfiguration);
```

Comment:
could we move validation to the `ClientConfig`? then we don't have to export too much